reviewfile.cs

```diff
@@ -0,0 +1,387 @@
+            
+            var view = new SGPrepOperationView(operation.Value);
+
+            if (view.EligibilitySubmission is null)
+            {
+                _logger.LogWarning("No eligibility submission on the view model for {OperationNumber}", opNumber);
+                return NotFound();
+            }
+
+            return Ok(new EligibilitySubmissionResponse
+            {
+                CirculationPeriod = new EligibilityCirculationPeriod
+                {
+                    StartDate = view.EligibilitySubmission.StartDate?.ToString("yyyy-MM-dd"),
+                    EndDate = view.EligibilitySubmission.EndDate?.ToString("yyyy-MM-dd"),
+                    Meeting = new EligibilityMeeting
+                    {
+                        Date = view.EligibilitySubmission.MeetingStartTime?.Date.ToString("yyyy-MM-dd"),
+                        EndTime = !view.EligibilitySubmission.EndTimeHasBeenSet ? null : (DateTime.Today + view.EligibilitySubmission.MeetingEndTime.Value.TimeOfDay).ToString("h:mm tt"),
+                        Justification = view.EligibilitySubmission.MeetingJustification,
+                        MeetingLink = view.EligibilitySubmission.MeetingLink,
+                        StartTime = !view.EligibilitySubmission.StartTimeHasBeenSet ? null : (DateTime.Today + view.EligibilitySubmission.MeetingStartTime.Value.TimeOfDay).ToString("h:mm tt")
+                    }
+                },
+                DocPreparationProcessing = view.EligibilitySubmission.IsIcapPaciRequired != null ?
+                    new EligibilityDocPreparationProcessing
+                    {
+                        IsIcapPaciRequired = view.EligibilitySubmission.IsIcapPaciRequired,
+                        ProcessingTrack = view.EligibilitySubmission.GetProcessingTrack()
+                    } : null
+            });
+        }
+
+        [HttpPut("{opNumber}/eligibility")]
+        [ProducesResponseType(StatusCodes.Status201Created)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        //[Authorize(SGPrepPolicies.CorrectOperationRolePolicy)]
+        public async Task<IActionResult> SaveEligibility([FromBody] EligibilitySubmissionRequest request, string opNumber)
+        {
+            ValidateUserContext();
+            
+            var streamId = opNumber.ToOperationStreamIdentifier();
+            var existingOperation = await _repo.GetOptionalAsync(streamId);
+
+            if (!existingOperation.HasValue)
+            {
+                return NotFound();
+            }
+
+            if (existingOperation.Value.EligibilitySubmission is not null)
+            {
+                throw new EligibilitySubmissionModelInvalidException("An eligibility submission already exists for this operation");
+            }
+
+            string resultValidation = string.empty;
+
+            switch (request.DocType)
+            {
+                case "PP":
```

Comment:
We could use consts inside these case statements instead of strings (avoid magic words)